### ISSUE TITLE
Most XML entities not being decoded by reader

### ISSUE THREAD

**Issue by kernwig** [oozcitak/xmlbuilder2#99]
**Describe the bug**
New XML entity decoding in v3.0 only works on &amp;amp; &amp;lt; and &amp;gt;, but not others. Interestingly, the XML builder does encode a few other characters like &amp;apos;.

**To Reproduce**

```
   const xmlResponse = create('<ssid>Me &amp; Myself&apos;s WiFi &copy;&#x1D306;</ssid>');
   const networkResponse = xmlResponse.end({format: 'object'});
   return networkResponse.ssid;
```

The output string is `Me &amp; Myself&apos;s WiFi &amp;copy;&amp;#x1D306;`

The first two entities were not touched, while the last two got double-encoded.

**Expected behavior**
Expected result to be `Me & Myself's WiFi ©𝌆`.

**Version:**
 - node.js: 12.16.1
 - xmlbuilder2: 3.0.1

**Additional context**
Related to #88

**Comment by oozcitak** [comment]
I'll take a look at `&apos;` it should be decoded as `'`. Also `&#x1D306;` should be decoded as `𝌆`. But `&` needs to be escaped when serialized as `&amp;` per [spec](https://www.w3.org/TR/DOM-Parsing/#dfn-concept-serialize-xml). `&copy;` is not an [XML predefined entity](https://www.w3.org/TR/xml/#sec-predefined-ent), so it should throw a parser exception. Browsers throw on undefined entities, e.g. with Chrome:

```js
const str = `<ssid>Me &amp; Myself&apos;s WiFi &copy;&#x1D306;</ssid>`;
const p = new DOMParser();
const s = new XMLSerializer();
s.serializeToString(p.parseFromString(str, "application/xml"));
```
```html
<ssid>
  <parsererror xmlns=\"http://www.w3.org/1999/xhtml\" style=\"display: block; white-space: pre; border: 2px solid #c77; padding: 0 1em 0 1em; margin: 1em; background-color: #fdd; color: black\">
    <h3>This page contains the following errors:</h3>
      <div style=\"font-family:monospace;font-size:12px\">
         error on line 1 at column 41: Entity 'copy' not defined\n
      </div>
    <h3>Below is a rendering of the page up to the first error.</h3>
  </parsererror>
</ssid>
``` 
without `&copy;`:
```js
const str = `<ssid>Me &amp; Myself&apos;s WiFi &#x1D306;</ssid>`;
const p = new DOMParser();
const s = new XMLSerializer();
s.serializeToString(p.parseFromString(str, "application/xml"));
```
```xml
<ssid>Me &amp; Myself's WiFi 𝌆</ssid>
```

**Comment by oozcitak** [comment]
Instead of throwing on unknown entities, I opted to silently ignore them.

**Comment by kernwig** [comment]
Thank you. My driving bug had to do with `&apos;`. When I was working around the bug in xmlbuilder2 v2.x, I used the **he** library and tried this larger variety of characters just to be thorough. Looks like you are now covering the "predefined entities" of XML, not the massive list of "standard" entities in HTML 5. [Reference](https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references)

**Comment by kernwig** [comment]
@oozcitak When might I expect the next release with this fix?

**Comment by oozcitak** [comment]
Oh shoot sorry. Forgot to release. I'll prepare one tomorrow. Thanks for reminding.

**Comment by oozcitak** [comment]
Released `v3.0.2` with the fix.

**Comment by kernwig** [comment]
I see that `<` and `>` also get re-encoded, but not `'`. 

```
   const xmlResponse = create('<ssid>Me &amp; Myself&apos;s &lt;WiFi&gt;&#x1D306;</ssid>');
   const networkResponse = xmlResponse.end({format: 'object'});
   return networkResponse.ssid;
```
*Expected* result: `Me & Myself's <WiFi>𝌆`

*Actual* result with v3.0.2: `Me &amp; Myself's &lt;WiFi&gt;𝌆`
